From 5133ca3b30d18bd4eeba7b0ecca0afb22edcd2f8 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Tue, 22 Sep 2020 12:24:31 -0700
Subject: [PATCH 1655/1851] v4l: xilinx: switch: Fix warnings for UNINIT

commit 01b04e3e345c935b8bb14d8b1aea27db7fb4f952 from
https://github.com/Xilinx/linux-xlnx.git

This patch fix the warnings for the UNINIT type.

The actual warnings are:

For xsw_probe():
Event var_decl: 	Declaring variable "subdev" without initializer
Event uninit_use: 	Using uninitialized value "subdev".

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-switch.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-switch.c b/drivers/media/platform/xilinx/xilinx-switch.c
index 354428f7bcfb..e249ac9b443c 100644
--- a/drivers/media/platform/xilinx/xilinx-switch.c
+++ b/drivers/media/platform/xilinx/xilinx-switch.c
@@ -391,7 +391,7 @@ static int xsw_probe(struct platform_device *pdev)
 	xsw->pads = devm_kzalloc(&pdev->dev, npads * sizeof(*xsw->pads),
 				 GFP_KERNEL);
 	if (!xsw->pads)
-		goto error;
+		goto error_resources;
 
 	for (i = 0; i < xsw->nsinks; ++i)
 		xsw->pads[i].flags = MEDIA_PAD_FL_SINK;
@@ -402,7 +402,7 @@ static int xsw_probe(struct platform_device *pdev)
 				    xsw->nsinks * sizeof(*xsw->formats),
 				    GFP_KERNEL);
 	if (!xsw->formats)
-		goto error;
+		goto error_resources;
 
 	for (i = 0; i < xsw->nsources; ++i)
 		xsw->routing[i] = i < xsw->nsinks ? i : -1;
@@ -436,6 +436,7 @@ static int xsw_probe(struct platform_device *pdev)
 
 error:
 	media_entity_cleanup(&subdev->entity);
+error_resources:
 	xvip_cleanup_resources(&xsw->xvip);
 	return ret;
 }
-- 
2.31.1

